frontend/Interfaces/buddy/DIP/ImageContainer.h

```diff
@@ -269,19 +201,18 @@ Img<T, N> &Img<T, N>::operator=(const Img<T, N> &m) {
 
 /**
  * @brief overload
- * @param rows Number of rows in a 2D array.
- * @param cols Number of columns in a 2D array.
+ * @param sizes Array of integers specifying an n-dimensional array shape.
  * @param data Pointer to the user data.
  * they just initialize the matrix header that points to the specified data.
  */
 template <typename T, size_t N>
-Img<T, N>::Img(int rows, int cols, T *data)
-    : MemRef<T, N>(), flags(0), dims(2), rows(rows), cols(cols) {
-  this->aligned = data;
-  this->sizes[0] = rows;
-  this->sizes[1] = cols;
-  this->size = total();
+Img<T, N>::Img(intptr_t *sizes, T *data) : MemRef<T, N>() {
```

Comment:
MemRef has constructor function `MemRef(const T *data, intptr_t sizes[N], intptr_t offset = 0);`. You can replace it.